packages/core/src/client.ts

```diff
@@ -2171,6 +2172,12 @@ export class MedplumClient extends EventTarget {
   ): Promise<Partial<BulkDataExport>> {
     const fhirPath = exportLevel ? `${exportLevel}/` : exportLevel;
     const url = this.fhirUrl(`${fhirPath}$export`);
+    let headers = options.headers as Record<string, string> | undefined;
+    if (!headers) {
```

Comment:
You can avoid this check if you move the `headers['Prefer'] = ...` line to after `addFetchOptionsDefaults`